### PR TITLE
Optional SAX Parser for Sitemaps

### PR DESCRIPTION
**Pull request by sebastian-nagel**
Provide a SAX-based parser for sitemaps, fixes #116
- use ```SiteMapParser sitemapParser = new SiteMapParserSAX(strict, allowPartialParse);```

### CODE REVIEW

**Comment by lewismc** [comment]
This looks like a really nice piece of code @sebastian-nagel 
Apart from some inconsistencies in the formatting of Javadoc, all I can see if that some new tests you've introduced are annotated as @ Ignore. Was that still intentional.
Nice work.

**Comment by sebastian-nagel** [comment]
Thanks, the Javadoc is now formatted.
Yes, the ignored tests are intentional: I've kept both parsers and the tests as similar as possible to compare them side by the side. That was also on the premise that the SAX-based parser will replace the DOM-based parser: move the class and test files, fix the class name, done. If we want to keep the DOM-based parser forever, we probably should avoid the duplicated code and also make the tests generic, so that we can run it with both parsers.

**Comment by lewismc** [comment]
ack

File: src/main/java/crawlercommons/sitemaps/SiteMapTester.java [hunk 1]
```diff
@@ -32,37 +32,50 @@
...
+            parse(url, mt, useSaxParser);
         }
     }
 
     /**
      * Parses a Sitemap recursively meaning that if the sitemap is a
      * sitemapIndex then it parses all of the internal sitemaps
      */
-    private static void parse(URL url, String mt) throws IOException, UnknownFormatException {
+    private static void parse(URL url, String mt, boolean useSaxParser) throws IOException, UnknownFormatException {
```
Comment (sebastian-nagel) [inline src/main/java/crawlercommons/sitemaps/SiteMapTester.java]: Ok, you're right, removing arguments can break things. I'll update the PR.
But I wouldn't call SiteMapTester an API (the "tester" indicates it): it's not thread-safe due to the usage of a static parser object. If we want to keep a stable API after removing the DOM-based parser and rename SiteMapParserSAX to SiteMapParser, a factory which is configured by properties, could be the better way.

Comment (jnioche) [inline src/main/java/crawlercommons/sitemaps/SiteMapTester.java]: > I wouldn't call SiteMapTester an API

Neither would I. I thought by looking through the commits that there was another change on the API proper. Not the case, as you pointed out.

Comment (sebastian-nagel) [inline src/main/java/crawlercommons/sitemaps/SiteMapTester.java]: Updated PR, usage of the SAX parser is now triggered by a Java property.

**Comment by jnioche** [comment]
thanks @sebastian-nagel. Could you please squash the commits? I think we're good to go after that.

**Comment by sebastian-nagel** [comment]
Commits until and including the merge commit (merge [matt-deboer-master](/crawler-commons/crawler-commons/tree/matt-deboer-master)) are squashed.

**Comment by sebastian-nagel** [comment]
Thanks!

**Pull request merged**